utils/utils.go

```diff
@@ -77,22 +76,27 @@ func CheckValidQuery(transferUrl *url.URL, isPlugin bool) (err error) {
 	query := transferUrl.Query()
 	_, hasRecursive := query["recursive"]
 	_, hasPack := query["pack"]
-	_, hasDirectRead := query["directread"]
+	directRead, hasDirectRead := query["directread"]
 
-	// If we are not the plugin, we should not use ?recursive (we should pass a -r flag)
+	// // If we are not the plugin, we should not use ?recursive (we should pass a -r flag)
 	if !isPlugin && hasRecursive {
-		return fmt.Errorf("Cannot use the recursive query parameter when not utilizing the pelican plugin")
+		return errors.New("cannot use the recursive query parameter when not utilizing the pelican plugin")
```

Comment:
On second thought, let's say "recursive query parameter not yet supported in URLs outside of plugin" or something to that effect. That tells the user it's coming, but not yet ready.

My second comment is that we need an issue to keep track of this. Can you open an issue and link it to this PR?

Reply:
This is the issue, I added it to it https://github.com/PelicanPlatform/pelican/issues/1165

The issue includes this PR and I have notes to modify this function as well as the tests. Anything else I need to do?

Reply:
Nope, thanks for linking it :)